src/umbraco.cms/Actions/ActionRestore.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using umbraco.interfaces;
+using umbraco.BasePages;
+
+namespace umbraco.BusinessLogic.Actions
+{
+	/// <summary>
+	/// This action is invoked when the content item is to be restored from the recycle bin
+	/// </summary>
+	public class ActionRestore : IAction
+	{
+		//create singleton
+#pragma warning disable 612,618
+		private static readonly ActionRestore m_instance = new ActionRestore();
```

Comment:
You don't need to ignore warnings for new code, just write the code so we don't need these things. m_instance should just be called '_instance'